rotkehlchen/chain/evm/decoding/gearbox/decoder.py

```diff
@@ -0,0 +1,173 @@
+        if not self.base.is_tracked(user_address := hex_or_bytes_to_address(user_address_bytes)):
+            return None
+
+        try:
+            pool_info = self.pools[context.tx_log.address]
+        except KeyError:
+            log.error(f'Could not find {self.evm_inquirer.chain_name} Gearbox pool info for the pool address {context.tx_log.address}')  # noqa: E501
+            return None
+
+        token = EvmToken(pool_info.farming_pool_token)
+        amount = token_normalized_value(
+            token_amount=hex_or_bytes_to_int(context.tx_log.data[:32]),
+            token=token,
+        )
+        shares = token_normalized_value(
+            token_amount=hex_or_bytes_to_int(context.tx_log.data[32:64]),
+            token=token,
+        )
+        return user_address, pool_info, amount, shares
+
+    def _get_note_by_pool(self, pool_info: GearboxPoolData, asset: Asset, amount: FVal) -> str:
+        """Determines if the user is providing liquidity or staking and returns
+        the appropriate note."""
+        action = 'providing liquidity' if asset.identifier in pool_info.lp_tokens else 'staking'
+        return f'Receive {amount} {asset.symbol_or_name()} after {action} in Gearbox'
+
+    def _decode_deposit(self, context: DecoderContext) -> DecodingOutput:
+        if (lp_data := self._decode_common(context, context.transaction.input_data[36:68])) is None:  # noqa: E501
+            return DEFAULT_DECODING_OUTPUT
+
+        user_address, pool_info, amount, shares = lp_data
+        for event in context.decoded_events:
+            if (
+                event.event_type == HistoryEventType.SPEND and
+                event.event_subtype == HistoryEventSubType.NONE and
+                event.balance.amount == amount and
+                event.location_label == user_address
+            ):
+                event.event_type = HistoryEventType.DEPOSIT
+                event.event_subtype = HistoryEventSubType.DEPOSIT_ASSET
+                event.counterparty = CPT_GEARBOX
+                event.notes = f'Deposit {event.balance.amount} {event.asset.symbol_or_name()} to Gearbox'  # noqa: E501
+                break
+
+        return DecodingOutput(
+            action_items=[
+                ActionItem(
+                    action='transform',
+                    from_event_type=HistoryEventType.RECEIVE,
+                    from_event_subtype=HistoryEventSubType.NONE,
+                    asset=Asset(asset_id),
+                    amount=shares,
+                    location_label=user_address,
+                    to_event_type=HistoryEventType.RECEIVE,
+                    to_event_subtype=HistoryEventSubType.RECEIVE_WRAPPED,
+                    to_notes=self._get_note_by_pool(pool_info, Asset(asset_id), shares),
+                    to_counterparty=CPT_GEARBOX,
+                ) for asset_id in pool_info.lp_tokens.union({pool_info.farming_pool_token})
```

Comment:
this logic is both for LPs and farms? if so state it in the docsrting. Also there is something that doesn't make much sense. Why is `lp_tokens` a set? Is it the underlying tokens? But if it is the anderlying tokens why you have receive_wrapped? And if it is the lp token why do you have multiple lp tokens for the pool?

Reply:
so some gearbox pools have multiple lp tokens, they are basically their old lp tokens.
lp_tokens is a set to avoid duplicates mainly